src/core/public/rendering/rendering_service.tsx

```diff
@@ -52,6 +52,7 @@ export class RenderingService {
           {chromeHeader}
           <AppWrapper chromeVisible$={chrome.getIsVisible$()}>
             <div className="app-wrapper-panel">
+              <div id="app-fixed-viewport" />
               <div id="globalBannerList">{bannerComponent}</div>
               <AppContainer classes$={chrome.getApplicationClasses$()}>{appComponent}</AppContainer>
```

Comment:
I understand the purpose, but I'm unsure why all this is necessary tbh. In 'stretch-to-height' mode apps, `app-wrapper-panel` (or even `.kibana-body`) already pretty much has the effective layout/positioning the added `#app-fixed-viewport` element has, isn't it?

Couldn't you just use `.app-wrapper-panel` or `.kbnBody` as your boundary element?

Reply:
The `.app-wrapper-panel` and `the `#kibana-body` have a height of 100% of the entire dashboard and they scroll with the dashboard itself, using that as boundary doesn't block the tooltip from being blocked right below the headers.

Reply:
@markov00 is correct, I tried using every top-level element that was available including:

- `document.body` or `.kbn-body` -- only at the top of the scroll height and moves when body is scrolled. See video below.
- `#kibana-body` -- full scroll height goes beyond viewport as soon as you scroll
- `.content` -- full scroll height goes beyond viewport as soon as you scroll
- `.app-wrapper` -- full scroll height goes beyond viewport as soon as you scroll
- `.app-wrapper-panel` -- full scroll height goes beyond viewport as soon as you scroll
- `.application` -- full scroll height goes beyond viewport as soon as you scroll

This was the last option unless someone else has any suggestions.

https://user-images.githubusercontent.com/19007109/110832139-97c9fb00-8260-11eb-91fb-6582b3774361.mp4

Reply:
> The .app-wrapper-panel and the #kibana-body` have a height of 100% of the entire dashboard and they scroll with the dashboard itself

Oh, I thought it was for a stretch to height app. You're right, for dashboard, the body is taking the effective height of the dashboard.